reactor-netty-core/src/main/java/reactor/netty/transport/AddressUtils.java

```diff
@@ -88,30 +88,48 @@ public static InetSocketAddress createUnresolved(String hostname, int port) {
 	}
 
 	/**
-	 * Parse unresolved InetSocketAddress. Numeric IP addresses will be detected and
-	 * resolved.
+	 * Parse unresolved InetSocketAddress. Numeric IP addresses will be detected and resolved.
 	 *
 	 * @param address ip-address or hostname
 	 * @param defaultPort the default port
-	 * @return InetSocketAddress for given parameters
+	 * @return {@link InetSocketAddress} for given parameters, only numeric IP addresses will be resolved
 	 */
 	public static InetSocketAddress parseAddress(String address, int defaultPort) {
+		return parseAddress(address, defaultPort, false);
+	}
+
+	/**
+	 * Parse unresolved InetSocketAddress. Numeric IP addresses will be detected and resolved.
+	 *
+	 * @param address ip-address or hostname
+	 * @param defaultPort is used when the host is parseable but not the port
+	 * @param strict if true throws an exception when the address cannot be parsed,
+	 * otherwise an unresolved {@link InetSocketAddress} is returned. It can include the case of the host
+	 * having been parsed but not the port (replaced by {@code defaultPort})
+	 * @return {@link InetSocketAddress} for given parameters, only numeric IP addresses will be resolved
+	 */
+	public static InetSocketAddress parseAddress(String address, int defaultPort, boolean strict) {
```

Comment:
I think the strict flag could probably be removed, even if the non-standard x-forwarded headers aren't clearly defined, an IPv6 address should have square brackets as per https://tools.ietf.org/html/rfc3986#section-3.2.2 so it can be used to construct URLs.

Reply:
This is a public API and it is not OK to change the behaviour.
Let me change the functionality for `X-Forwarded` in the `reactor-netty-http` module

Reply:
It could also be seen as tightening the validation, i.e. the behavior that should be, that is if you want to avoid adding a new API option that might not be necessary (or could be added later if it turned out to be essential).

Reply:
I prefer to expose a new method as this method for parsing the address is not directly connected to `Forwarded`/`X-Forwarded` and somebody may rely on the behaviour.